core/src/main/scala/caliban/schema/RootType.scala

```diff
@@ -10,9 +11,17 @@ case class RootType(
   additionalTypes: List[__Type] = List.empty,
   additionalDirectives: List[__Directive] = List.empty
 ) {
+  private val primitiveTypes: List[__Type] = List(
```

Comment:
Is it possible to include only those that are actually used in other parts of the schema?

Reply:
Ah, I see the problem you mentioned though. Maybe instead of adding these to the schema, just add them during the validation?

Reply:
Good idea, I changed that in 102a38a!